pkg/sloop/processing/relationship.go

```diff
@@ -0,0 +1,126 @@
+package processing
+
+import (
+	"encoding/json"
+	"github.com/dgraph-io/badger/v2"
+	"github.com/golang/glog"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/pkg/errors"
+	"github.com/salesforce/sloop/pkg/sloop/kubeextractor"
+	"github.com/salesforce/sloop/pkg/sloop/store/typed"
+	"github.com/salesforce/sloop/pkg/sloop/store/untyped"
+	"github.com/salesforce/sloop/pkg/sloop/store/untyped/badgerwrap"
+)
+
+
+
+
+//  transfer watchRec.Payload to Relationship payload: Convert watchRec to record in relationship table and save key-value pair
+func updateRelationshipTable(tables typed.Tables, txn badgerwrap.Txn, watchRec *typed.KubeWatchResult, metadata *kubeextractor.KubeMetadata) error {
+	if watchRec.Kind != kubeextractor.PodKind {
+		glog.Infof("Skipping relationship table update as kubewatch result is not PodKind")
+		return nil
+	}
+	// convert node -> pod relationship from watchRec
+	// get new relationship record from watch record
+	record, err := getRelationshipRec(watchRec)
+
+	//convert timestamp
+	ts, err := ptypes.Timestamp(watchRec.Timestamp)
+	if err != nil {
+		return errors.Wrap(err, "could not convert timestamp")
+	}
+    // get new key
+	key := typed.NewRelationshipKey(ts, watchRec.Kind, metadata.Namespace, metadata.Name, metadata.Uid).String()
+	// get existing value for the new key
+	value, err := getRelationshipValue(tables, txn, key, metadata, watchRec)
+	if err != nil {
+		return errors.Wrapf(err, "could not get record for key %v", key)
+	}
+	// append new record
+	value.Relationships = append(value.Relationships, record)
+	// save back to DB
+	err = tables.RelationshipTable().Set(txn, key, value)
+	if err != nil {
+		return errors.Wrapf(err, "put for the key %v failed", key)
+	}
+
+	return nil
+}
+
+// return:
+// record payload entry, error if any
+func getRelationshipRec(watchRec *typed.KubeWatchResult) (string, error) {
```

Comment:
consider the name `buildRelationshipRec`

Reply:
updated